packages/babel-plugin/src/next.tsx

```diff
@@ -0,0 +1,60 @@
+import * as t from '@babel/types';
+import { declare } from '@babel/helper-plugin-utils';
+import { importSpecifier } from './utils/ast-builders';
+import { visitCssPropPath } from './css-prop';
+import { visitStyledPath } from './styled';
+import { State } from './types';
+
+export default declare<State>((api) => {
+  api.assertVersion(7);
+
+  return {
+    inherits: require('babel-plugin-syntax-jsx'),
+    visitor: {
+      ImportDeclaration(path, state) {
+        if (path.node.source.value === '@compiled/css-in-js') {
+          state.compiledImportFound = true;
+          path.node.specifiers = path.node.specifiers
+            .filter(
+              (specifier) =>
+                specifier.local.name !== 'styled' && specifier.local.name !== 'ClassNames'
+            )
+            .concat([importSpecifier('CC'), importSpecifier('CS')]);
+        }
+      },
+      VariableDeclaration(path, state) {
+        if (!state.declarations) {
+          state.declarations = {};
+        }
+
+        if (!t.isIdentifier(path.node.declarations[0].id)) {
+          return;
+        }
+
+        const declarationName = path.node.declarations[0].id.name;
+        state.declarations[declarationName] = path.node;
+      },
+      TaggedTemplateExpression(path, state) {
+        if (!state.compiledImportFound) {
+          return;
+        }
+
+        visitStyledPath(path, state);
+      },
+      CallExpression(path, state) {
+        if (!state.compiledImportFound) {
+          return;
+        }
+
+        visitStyledPath(path, state);
+      },
+      JSXOpeningElement(path, state) {
```

Comment:
@madou: Does spread operator works here 🤔? Something like this:

```
import '@compiled/css-in-js';

const props = {
  css: 'font-size:12px;'
}

const MyDiv = () => {
  return <div {...props}>hello</div>
};
```

Reply:
Didn't find `JSXSpreadAttribute` 🤔.

Reply:
hi - yeah currently that isn't supported. i don't see that as a very common use case. do you think it's useful to add it in?

my thoughts: do the bare minimum and then add more if it has wide spread usage.

needing a static `css` prop directly on the jsx element seems like the right path for me.

Reply:
Nah!! not that much important for `css` prop tbh. But yups for other props we should definitely consider it.

> my thoughts: do the bare minimum and then add more if it has wide spread usage.

^^ I am sold 😛